orttraining/orttraining/python/training/checkpoint.py

```diff
@@ -168,114 +183,80 @@ def __init__(self, checkpoint_files, clean_state_dict=None):
+        param_name = param_name.split('_fp16')[0]
+        return param_name, optimizer_key, view_num, fp16_key
 
     def _update_weight_statistics(self, name, value):
-        self.weight_shape_map[name] = value.size()  # original shape of tensor
-
-    def _reshape_tensors(self, state_dict, fp16):
-        for k, v in state_dict.items():
-            if k.startswith('Moment_'):
-                _, weight_name, _ = self._split_moment_name(k)
-                set_size = self.weight_shape_map[weight_name]
-                state_dict[k] = v.reshape(set_size)
-                state_dict[weight_name] = state_dict[weight_name].reshape(set_size)
-        return state_dict
+        if name not in self.weight_shape_map:
+            self.weight_shape_map[name] = value.size() #original shape of tensor
+
+    def _reshape_tensor(self, key):
+        value = self.aggregate_state_dict[key]
+        weight_name, _, _, _ = self._split_name(key)
+        set_size = self.weight_shape_map[weight_name]    
+        self.aggregate_state_dict[key] = value.reshape(set_size)
+     
+    def _aggregate(self, param_dict):
+        for k,v in param_dict.items():
+            weight_name, optimizer_key, view_num, fp16_key = self._split_name(k)
+            if view_num is not None:
+                # parameter is sharded
+                param_name = optimizer_key + weight_name + fp16_key
+                
+                if param_name in self.aggregate_state_dict and optimizer_key not in ['Update_Count_']:
+                    self.sharded_params.add(param_name)
+                    # Found a previous shard of the param, concatenate shards ordered by ranks
+                    self.aggregate_state_dict[param_name] = torch.cat((self.aggregate_state_dict[param_name], v))
+                else:
+                    self.aggregate_state_dict[param_name] = v            
+            else:
+                if k in self.aggregate_state_dict:
+                    assert (self.aggregate_state_dict[k] == v).all(), "Unsharded params must have the same value"
+                else:
+                    self.aggregate_state_dict[k] = v
+                self._update_weight_statistics(weight_name,v)
 
     def aggregate_checkpoints(self):
-        checkpoint_dir = os.path.dirname(self.checkpoint_files[0])
+        checkpoint_dir=os.path.dirname(self.checkpoint_files[0])
         checkpoint_prefix = self.checkpoint_files[0].split('.ZeRO')[0]
-        self.aggregate_state_dict = dict()
+        self.aggregate_state_dict=dict()
 
-        is_fp16 = False
-        weight_offset = dict()
         for i in range(self.world_size):
             checkpoint_name = _get_checkpoint_name(checkpoint_prefix, True, i, self.world_size)
             rank_state_dict = torch.load(checkpoint_name, map_location=torch.device("cpu"))
             if 'model' in rank_state_dict:
                 rank_state_dict = rank_state_dict['model']
-
+            
```

Comment:
Check code style for the python files otherwise `flake8` will raise a failure in the CI